### PR TITLE
serie: init at 0.1.0

### PR DESCRIPTION
**Pull request by matthiasbeyer**
## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by matthiasbeyer** [comment]
Can someone tell me what the actual formatting error is?

**Comment by Gliczy** [comment]
You need to format your file using `nixfmt`.

**Comment by matthiasbeyer** [comment]
Well I already did that (see the force push)...

**Comment by uncenter** [comment]
I'm getting the same formatting result. Make sure you are using `nixfmt-rfc-style` specifically.

File: pkgs/by-name/se/serie/package.nix [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  rustPlatform,
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "serie";
+  version = "0.1.0";
+
+  src = fetchFromGitHub {
+    owner = "lusingander";
+    repo = pname;
```
Comment (uncenter) [inline pkgs/by-name/se/serie/package.nix:13]:
```suggestion
    repo = "serie";
```
If I recall correctly `repo = pname;` is a bad practice.

**Review by uncenter — CHANGES_REQUESTED** [review]

File: pkgs/by-name/se/serie/package.nix (hunk 1 above)
Comment (matthiasbeyer) [reply pkgs/by-name/se/serie/package.nix]: You're absolutely right.

**Comment by uncenter** [comment]
Result of `nixpkgs-review pr 330843` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>serie</li>
  </ul>
</details>

**Comment by uncenter** [comment]
Looks like you need the Darwin `CoreGraphics` framework. https://github.com/nix-community/nix-init can detect and add these automatically FYI.

**Comment by Gliczy** [comment]
Result of `nixpkgs-review pr 330843` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>serie</li>
  </ul>
</details>

**Review by Gliczy — APPROVED** [review]
LGTM on the linux side.

**Comment by uncenter** [comment]
Not sure if this is the reason for the error but I typically see darwin framework build inputs added with:
```nix
  stdenv.isDarwin (
      with darwin.apple_sdk.frameworks;
      [
        CoreGraphics
      ]
    );

**Comment by uncenter** [comment]
Result of `nixpkgs-review pr 330843` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>serie</li>
  </ul>
</details>

File: pkgs/by-name/se/serie/package.nix [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  src = fetchFromGitHub {
+    owner = "lusingander";
+    repo = "serie";
+    rev = "v${version}";
+    hash = "sha256-81DXu5SSL5TWtnIDBWCWlatRtEL00h4vOnRA0HuTInQ=";
+  };
+
+  cargoHash = "sha256-Vx3UGsWvudW33CFGU8rS+QCANxG7aXBf0zkYJxcUVps=";
+
+  buildInputs = lib.optionals stdenv.isDarwin (with darwin.apple_sdk.frameworks; [ CoreGraphics ]);
```
Comment (uncenter) [inline pkgs/by-name/se/serie/package.nix:22]: Sorry, looks like it also needs `AppKit`!

**Review by uncenter — CHANGES_REQUESTED** [review]

File: pkgs/by-name/se/serie/package.nix (hunk 2 above)
Comment (uncenter) [reply pkgs/by-name/se/serie/package.nix:22]: I've tested locally and can confirm that should be all it needs, building for me on darwin.

**Comment by uncenter** [comment]
https://github.com/lusingander/serie/releases/tag/v0.1.1 is out.

**Comment by uncenter** [comment]
Result of `nixpkgs-review pr 330843` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>serie</li>
  </ul>
</details>

**Review by uncenter — APPROVED** [review]

**Comment by matthiasbeyer** [comment]
Not going to self-merge! :wink:

**Comment by matthiasbeyer** [comment]
Result of `nixpkgs-review pr 330843` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>nixpkgs-manual</li>
    <li>serie</li>
  </ul>
</details>

**Pull request merged**